packages/core/src/util/configNormalizers/ignoreEndpoints.js

```diff
@@ -110,6 +111,39 @@ exports.fromEnv = function fromEnv(ignoreEndpointsEnv) {
   }
 };
 
+/**
+ * Reads and normalizes the `INSTANA_IGNORE_ENDPOINTS_PATH` configuration from a YAML file.
+ *
+ * - The YAML file must have a root key of either `tracing` or `com.instana.tracing`.
+ * - If `com.instana.tracing` is detected, it will be logged and automatically mapped to `tracing`.
+ * - Returns an empty object if the file is missing, malformed, or improperly structured.
+ *
+ * @param {string} yamlFilePath
+ */
+exports.fromYaml = function fromYaml(yamlFilePath) {
+  /** @type {Record<string, any>} */
+  const endpointsConfig = read(yamlFilePath);
+  if (!endpointsConfig || typeof endpointsConfig !== 'object') {
+    logger?.debug(`Invalid YAML content. Expected an object, but got: ${typeof endpointsConfig}`);
+    return {};
+  }
+
+  if (!endpointsConfig.tracing && !endpointsConfig['com.instana.tracing']) {
+    logger?.debug('Invalid YAML structure. The root key must be "tracing".');
+    return {};
+  }
+
+  if (endpointsConfig['com.instana.tracing']) {
+    logger?.info(
+      // eslint-disable-next-line max-len
+      `Detected the root key "com.instana.tracing" in the YAML file at "${yamlFilePath}". This format is accepted, but please migrate to using "tracing" as the root key.`
+    );
+    endpointsConfig.tracing = endpointsConfig['com.instana.tracing'];
+    delete endpointsConfig['com.instana.tracing'];
+  }
+  return exports.normalizeConfig(endpointsConfig.tracing['ignore-endpoints']) || {};
```

Comment:
qs: Why do we need "|| {};"? normalizeConfig should return {} if anything is broken.